tests/test_workflow_api.py

```diff
@@ -0,0 +1,31 @@
+import pytest
+from devtools import debug
+from sqlmodel import select
+
+from fractal_server.app.models import Task
+from fractal_server.app.models import TaskCreate
+from fractal_server.tasks import collect_tasks
+
+async def test_workflow_post(db, client, MockCurrentUser):
+    fake_workflow = {
+        "name": "My Workflow",
+        "resource_type": "core task",
+        "input_type": "Boh",
+        "output_type": "Boh",
+    }
+    async with MockCurrentUser(persist=True):
+        
+        res1 = await client.post(
+            "api/v1/workflow/",
+            json=fake_workflow,
+        )
+        res2 = await client.post(
+            "api/v1/workflow/",
+            json=fake_workflow,
+        )
+        assert res1.status_code==201
+        assert res2.status_code==422
```

Comment:
these assertion are weak. you should test that the object is actually persisted to the db and that the relationship with a project is actually established.